app/src/main/java/com/randomappsinc/studentpicker/home/HomeActivity.java

```diff
@@ -93,6 +95,12 @@ public void onSlide(@NonNull View bottomSheet, float slideOffset) {
         preferencesManager.increaseNumAppOpens();
         if (preferencesManager.getNumAppOpens() == NUM_APP_OPENS_FOR_RATING_ASK) {
             showPleaseRateDialog();
+        } else if (preferencesManager.getNumAppOpens() % NUM_APP_OPENS_FOR_RESTORE_AND_BACKUP == 0 &&
+                       !preferencesManager.hasSeenBackupAndRestore() &&
+                            (preferencesManager.getBackupUri() == null
+                                    || preferencesManager.getBackupUri().isEmpty())) {
+            preferencesManager.onBackupAndRestoreSeen(true);
```

Comment:
Very small detail but this line should be in the method as they should be coupled together. You don't want someone calling the method without running this line of code.